src/drivers/interrupt/cortexm/cortexm_nvic.c

```diff
@@ -178,18 +177,14 @@ extern void __pending_handle(void);
 static void fill_irq_saved_ctx(struct irq_saved_state *state,
                uint32_t *opaque, struct irq_enter_ctx *regs) {
 	struct exc_saved_base_ctx *ctx;
+	size_t offset;
 
 	ctx = (struct exc_saved_base_ctx *) opaque;
 
-	if (!interrupted_from_fpu_mode(regs->lr)) {
-		memcpy(&state->ctx, ctx, BASE_CTX_SIZE);
-		state->misc[0] = regs->lr;
-		state->misc[1] = regs->sp;
-	} else {
-		memcpy(&state->ctx, ctx, FPU_CTX_SIZE);
-		state->misc[18] = regs->lr;
-		state->misc[19] = regs->sp;
-	}
+	/* Save LR and SP below the context. */
+	offset = interrupted_from_fpu_mode(regs->lr) ? FPU_CTX_SIZE : 0;
```

Comment:
There's no `memcpy()` now, is it correct?

Reply:
Yes, since we do not need registers values that were before the interrupt happened.

Reply:
We do not need them inside interrupt handling

Reply:
Did you check it on STM32 board? If it works, I think it's ok

Reply:
Yes, I have checked that before and checked right now after `git rebase`